### PR TITLE
Added Parameters section to swagger:route

### PR DESCRIPTION
**Pull request by jucardi**
Working on a company project, we have a custom version of the Swagger API which allow us to do calls to our microservices by reading the swagger documentation generated from our builds.

We use Gin for our API endpoints in our Go Microservices and unfortunately we were unable to tie the Request models to the router definitions.

I added a section in the `swagger:route` parser, so `Parameters` can be included with their properties, to easily allow adding a reference to a `swagger:model` for body params. Multiple parameters can be added under `Parameters` by simply using the `+` prefix when a new parameter starts. Example:

```
//  Create a pet based on the parameters.
//
//  Consumes:
//  - application/json
//  - application/x-protobuf
//  
//  Produces:
//  - application/json
//  - application/x-protobuf
//
//  Schemes: http, https, ws, wss
//
//  Parameters:
//  + name: request
//    in: body
//    schema: modelName
//  + name: id
//    in: path
//
//  Responses:
//    default: body:genericError
//    200: body:someResponse
//    422: body:validationError
//
//  Security:
//    api_key:
//    oauth: read, write
```

### CODE REVIEW

**Comment by jucardi** [comment]
I'll increase the coverage in a bit

File: generator/bindata.go [hunk 1]
```diff
@@ -729,34 +729,34 @@ func AssetNames() []string {
 // _bindata is a table, holding each asset generator, mapped to its name.
 var _bindata = map[string]func() (*asset, error){
```
Comment (jucardi) [inline generator/bindata.go]: The changes in this file are just the result of a `gofmt -s -w -l .`, no changes in logic. If needed they could be rolled back before merging

**Comment by casualjim** [comment]
have you seen swagger:operation it allows you to specify routes but with the full swagger yaml syntax. 

This PR somewhat duplicates this, but with bespoke syntax. I wonder where the type goes and some of the other validations etc that can be defined on parameters.
It's great that you figured out how to extend this :100:

**Comment by casualjim** [comment]
O I hadn't seen your response on #1404 
I see you have "bigger" plans, I'm good with your remarks there would you be up for making this a bit more complete so that it covers most of the possible use cases?

**Comment by jucardi** [comment]
Hi @casualjim will do, I'll cover more validations, such as only allow the use of schemas when `in:body` or allow `allowEmptyValue` to be set only if `in:query` or `in:formData`.

Was there any other use cases you had in mind?

**Comment by casualjim** [comment]
once you have the ability to specify types you probably also want the ability to specify the validations they have to pass. Things min length, pattern, min/max etc come to mind.
Feel free to ping me on slack https://slackin.goswagger.io

**Comment by jucardi** [comment]
the constraints, you're absolutely right, I completely forgot about them, I'll work on those as well

**Comment by jucardi** [comment]
@casualjim I added a section to parse the schema in parameters, covers `max`, `min`, `maxLength`, `minLength`, `default`, `enum`, `format`, it also does type assertion when using these constraints to ensure they are assigned to the right types.

Let me know if you have any observations. Thanks!

**Pull request merged**